packages/components/src/templates/next/components/internal/Siderail/SiderailLabel.tsx

```diff
@@ -0,0 +1,57 @@
+import { BiRightArrowAlt } from "react-icons/bi"
+import { tv } from "tailwind-variants"
+
+import type { Item } from "./types"
+import { useSiderailContext } from "./SiderailContext"
+
+const createSiderailLabelStyles = tv({
+  slots: {
+    container:
+      "group flex gap-1.5 px-3 py-4 transition hover:text-brand-interaction",
+    icon: "h-6 w-6 opacity-0 transition group-hover:translate-x-1 group-hover:opacity-100",
+  },
+  variants: {
+    isCurrent: {
+      true: {
+        container: "text-brand-interaction",
+      },
+    },
+  },
+})
+
+const compoundStyles = createSiderailLabelStyles()
+
+interface SiderailLabelProps extends Omit<Item, "childPages"> {
+  className?: string
+  showIconOnHover?: boolean
+}
+
+export const SiderailLabel = ({
+  isCurrent,
+  title,
+  url,
+  className,
+  showIconOnHover,
+}: SiderailLabelProps) => {
+  const { LinkComponent } = useSiderailContext()
+  if (isCurrent) {
+    return (
+      <p className={compoundStyles.container({ className, isCurrent })}>
+        {title}
+      </p>
+    )
+  }
+
+  return (
+    <LinkComponent
+      href={url}
+      className={compoundStyles.container({ className })}
+    >
```

Comment:
nit: related to the issue on arrow positioning, I think we should capture the user's arrow on the entire width of the siderail, not just for the text?

<img width="292" alt="Screenshot 2024-07-31 at 17 34 43" src="https://github.com/user-attachments/assets/efb88fd0-120f-4be8-aed1-d459f36fae0d">

as compared to 
<img width="281" alt="Screenshot 2024-07-31 at 17 34 52" src="https://github.com/user-attachments/assets/859898c5-8bd1-4ee3-96ec-65f61a62dd46">

Reply:
made the arrow inline muaha. but the touch target will still be like that la it's flex wrap